### PR TITLE
T102 deploy using terraform

### PR DESCRIPTION
**Pull request by zyusuf88**
Terraform code organised  into modules for ECS, network, and Cloudflare DNS
also  .gitignore to exclude terraform files

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,4 +36,43 @@ next-env.d.ts
...
+override.tf
+override.tf.json
+*_override.tf
+*_override.tf.json
+
+# Ignore transient lock info files created by terraform apply
+.terraform.tfstate.lock.info
+
+# Include override files you do wish to add to version control using negated pattern
+# !example_override.tf
```
Comment (KaltunAhmed) [inline .gitignore:71]: Do we need to uncomment this?

File: terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
+
+module "network" {
+  source = "./modules/network"
+  vpc_cidr      = var.vpc_cidr
+  subnet_1_cidr = var.subnet_1_cidr
+  subnet_2_cidr = var.subnet_2_cidr
+}
```
Comment (KaltunAhmed) [inline terraform/main.tf:7]: Question: Do we need to add a NAT gateway 🧐

Comment (zyusuf88) [reply terraform/main.tf:7]: 😭 i’m scared to touch the code again

File: .gitignore (hunk 1 above)
Comment (zyusuf88) [reply .gitignore:71]: I mean not necessarily. I don’t have the file anyways

File: terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  default     = "10.0.2.0/24"
+}
+
+variable "zone_name" {
+  description = "Route 53 hosted zone name"
+  type        = string
+  default     = "zeynabyusuf.com"
+}
+
+# variable "domain_name" {
```
Comment (KaltunAhmed) [inline terraform/variables.tf]: Do we need to remove these comments?

File: .gitignore (hunk 1 above)
Comment (KaltunAhmed) [reply .gitignore:71]: Ahhh so we can remove it and then add it back when we need it?

**Review by KaltunAhmed — DISMISSED** [review]
Looks good to me!

We can merge it 🚀

File: terraform/main.tf (hunk 2 above)
Comment (zyusuf88) [reply terraform/main.tf:7]: Also I only have public subnets so a NAT gateway wouldn’t  add additional security if that’s what you were thinking.

definitely if i had a private subnet

Comment (KaltunAhmed) [reply terraform/main.tf:7]: Okay nice, yh i was just wondering

File: .gitignore (hunk 1 above)
Comment (zyusuf88) [reply .gitignore:71]: yeah!

File: terraform/variables.tf (hunk 3 above)
Comment (zyusuf88) [reply terraform/variables.tf]: oh yes good spot!!

**Review by KaltunAhmed — APPROVED** [review]

**Pull request merged**